tests/integration/test_relation.py

```diff
@@ -103,17 +104,25 @@ async def test_delete_unused_ingresses(ops_test: OpsTest, app_name: str):
     assert isinstance(ops_test.model, Model)
     model_name = ops_test.model_name
 
-    def compare_svc_hostnames(expected: List[str]) -> bool:
-        all_ingresses = api_networking.list_namespaced_ingress(namespace=model_name)
-        return expected == [ingress.spec.rules[0].host for ingress in all_ingresses.items]
-
-    assert compare_svc_hostnames(["any"])
+    def assert_svc_hostnames(expected: List[str], timeout=300):
+        time_start = time.time()
+        while True:
+            all_ingresses = api_networking.list_namespaced_ingress(namespace=model_name)
+            try:
+                assert expected == [ingress.spec.rules[0].host for ingress in all_ingresses.items]
+                break
+            except AssertionError:
+                if time.time() - time_start > timeout:
+                    raise
+                time.sleep(1)
+
+    assert_svc_hostnames(["any"])
     await ops_test.juju("config", INGRESS_APP_NAME, "service-hostname=new-name")
     await ops_test.model.wait_for_idle(status="active")
-    assert compare_svc_hostnames(["new-name"])
+    assert_svc_hostnames(["new-name"])
```

Comment:
Should this use a tuple since it is constant?

Reply:
As I mentioned above, if not using the tuple inside `assert_svc_hostnames`, then using a tuple as an argument requires conversion.